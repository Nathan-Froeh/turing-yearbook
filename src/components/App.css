html {
  height: 100%;
  overflow: hidden;
  box-sizing: border-box;
}

body {
  background: url('https://ak9.picdn.net/shutterstock/videos/1007204479/thumb/1.jpg');
  background-size: cover;
  background-attachment: fixed;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
}

.App {
  text-align: center;
}

h1 {
  margin: 0;
  background-color: rgba(255, 255, 255, 0.95);
  padding: 10px;
  color: limegreen;
}

.selectors {
  display: flex;
  justify-content: flex-end;
  align-content: center;
  padding-right: 40px;
}

.group-selector {
  border-radius: 5px;
  margin-right: 10px;
}

#add-student-icon {
  height:30px;
  width: 30px;
}
