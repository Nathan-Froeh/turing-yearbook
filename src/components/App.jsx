import React, { Component } from 'react';
import Cohort from './Cohort';
import people from '../data/yearbook-data.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      group: people.staff
    }
  }

  setGroup = (group) => {
    if(group === 'staff') {
      this.setState({group: people.staff})
    } else {
      this.setState({group: people.students})
    }
  }

  render() {
    console.log('Render App')
    return (
      <div className="App">
        <header className="App-header">
        <h1>Turing Yearbook</h1>
        <button onClick={()=>this.setGroup('staff')}>Staff</button>
        <button onClick={()=>this.setGroup('students')}>Students</button>
        </header>
        <Cohort group={this.state.group}/>
      </div>
    );
  }
}

export default App;
