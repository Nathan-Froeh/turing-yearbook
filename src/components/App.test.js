import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';
import people from '../data/yearbook-data.js';


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App/>)
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders with all components in order', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('setGroup should change the group', () => {
    const expected1 = {
      group: people.staff,
      groupType: 'Staff',
      isHidden: true
    }
    const expected2 = {
      group: people.students,
      groupType: 'Students',
      isHidden: true
    }
    expect(wrapper.state()).toEqual(expected1)
    wrapper.instance().setGroup('students')
    expect(wrapper.state()).toEqual(expected2)
  })

  it('handleAdd should toggle isHidden', () => {
    expect(wrapper.state('isHidden')).toEqual(true)
    wrapper.instance().handleAdd()
    expect(wrapper.state('isHidden')).toEqual(false)
  })

})
