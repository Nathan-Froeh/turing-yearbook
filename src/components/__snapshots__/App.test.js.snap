// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with all components in order 1`] = `ShallowWrapper {}`;
